xarray/core/indexes.py

```diff
@@ -1017,6 +1017,13 @@ def stack(
     def unstack(self) -> tuple[dict[Hashable, Index], pd.MultiIndex]:
         clean_index = remove_unused_levels_categories(self.index)
 
+        if not clean_index.is_unique:
+            raise ValueError(
+                "Cannot unstack a non-unique MultiIndex. To restore the previous "
```

Comment:
Possibly this will be confusing in re "previous behavior" for most who hit this and we could just suggest the drop duplicates?